src/core/sys/posix/ucontext.d

```diff
@@ -921,19 +921,125 @@ else version (CRuntime_Musl)
-        sigset_t    uc_sigmask;
-        stack_t     uc_stack;
-        ucontext*   uc_link;
-        size_t      uc_mcsize;
-        mcontext_t* uc_mcontext;
+        version (X86_Any)
+        {
+            struct __darwin_mmst_reg
+            {
+                char[10] __mmst_reg;
+                char[6] __mmst_rsrv;
+            }
+
+            struct __darwin_xmm_reg
+            {
+                char[16] __xmm_reg;
+            }
+
+            struct __darwin_exception_state
+            {
+                uint __trapno;
+                uint __err;
+                version (X86_64)
+                    ulong __faultvaddr;
+                else
+                    uint __faultvaddr;
+            }
+
+            struct __darwin_thread_state
+            {
+                version (X86_64)
+                    ulong[21] __regs;
+                else
+                    uint[16] __regs;
+            }
+
+            struct __darwin_float_state
+            {
+                int[2] __fpu_reserved;
+                ushort __fpu_fcw;
+                ushort __fpu_fsw;
+                uint[7] __fpu_regs;
+                __darwin_mmst_reg[8] __fpu_stmm;
+                version (X86_64)
+                {
+                    __darwin_xmm_reg[16] __fpu_xmm;
+                    char[6*16] __fpu_rsrv4;
+                }
+                else
+                {
+                    __darwin_xmm_reg[8] __fpu_xmm;
+                    char[14*16] __fpu_rsrv4;
+                }
+                int __fpu_reserved1;
+            }
+
+            struct __darwin_mcontext
+            {
+                __darwin_exception_state __es;
+                __darwin_thread_state __ss;
+                __darwin_float_state __fs;
+            }
+        }
+        else version (PPC_Any)
+        {
+            struct __darwin_exception_state
+            {
+                uint __dar;
+                uint __dsisr;
+                uint __exception;
+                uint __pad0;
+                uint[4] __pad1;
+            }
+
+            struct __darwin_thread_state
+            {
+                uint[40] __regs;
+            }
+
+            struct __darwin_float_state
+            {
+                double[32] __fpregs;
+                uint __fpscr_pad;
+                uint __fpscr;
+            }
+
+            struct __darwin_vector_state
+            {
+                uint[32][4] __save_vr;
+                uint[4] __save_vscr;
+                uint[4] __save_pad5;
+                uint __save_vrvalid;
+                uint[7] __save_pad6;
+            }
+
+            struct __darwin_mcontext
+            {
+                __darwin_exception_state __es;
+                __darwin_thread_state __ss;
+                __darwin_float_state __fs;
+                __darwin_vector_state __vs;
+            }
+        }
     }
 
-    alias ucontext_t = ucontext;
+    static if (__traits(compiles, __darwin_mcontext))
```

Comment:
This, because no other targets have ucontext.h other than x86 and PPC (it got deprecated before ARM was introduced).

Reply:
As far as I can see, the functions are deprecated, but not the types [1].

[1] https://github.com/apple/darwin-xnu/blob/a449c6a3b8014d9406c2ddbdc81795da24aa7443/bsd/arm/_mcontext.h#L29

Reply:
They really don't make finding these definitions easy. I couldn't see it on opensource.apple.com :-)

Reply:
Yeah, you need to know where to look. GitHub makes that easier.

You really need to have these files on your system and the latest version of the SDK. It will make it much easier. You can create a free Apple ID and download Xcode here: https://developer.apple.com/download/more.